File: src/1512. Number of Good Pairs/goodpairs.java
/*
Solution to 1512. Number of Good Pairs(https://leetcode.com/problems/number-of-good-pairs/description/)
*/



class Solution {
    public int numIdenticalPairs(int[] nums) {

        int cnt = 0;
        int[] freq = new int[101]; //cos max size is 100 given
        for (int i: nums){
            cnt += freq[i]++;
        }
        return cnt;
        
    }
}

File: src/1512. Number of Good Pairs/goodpairs.py
##Solution to 1512. Number of Good Pairs(https://leetcode.com/problems/number-of-good-pairs/description/)


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        cnt = 0
        freq ={}

        for i in nums:
            if i in freq:
                cnt += freq[i]
                freq[i] += 1
            else:
                freq[i] = 1
        return cnt
            